includes/base/widget-base.php

```diff
@@ -371,12 +372,21 @@ protected function get_initial_config() {
 			'upsale_data' => $this->get_upsale_data(),
 		];
 
+		$config['upsale_data'] = apply_filters( 'elementor/widgets/' . $this->get_name() . '/custom_promotion', $config['upsale_data'] ) ?? $this->get_upsale_data();
```

Comment:
@TzviRabinovitch @Barcov 
Why do you have multiple calls to `$this->get_upsale_data()`?

Reply:
Looks like a redundent call

Reply:
What if someome unsets the variable using the filter? Then we fallback to get_upsale_data() because we never want the variable to be empty.

Reply:
So, store the original outside the filter. Why not call multiple times?

Reply:
Understood.
@Barcov you can tweak this in ED-13558